src/Microsoft.NetCore.Analyzers/Core/Security/DoNotCallDangerousMethodsInDeserialization.cs

```diff
@@ -0,0 +1,219 @@
+                    var serializableAttributeTypeSymbol = WellKnownTypes.SerializableAttribute(compilation);
+                    var streamingContextTypeSymbol = WellKnownTypes.StreamingContext(compilation);
+
+                    if (serializableAttributeTypeSymbol == null || streamingContextTypeSymbol == null)
+                    {
+                        return;
+                    }
+
+                    var builder = ImmutableArray.CreateBuilder<IMethodSymbol>();
+                    var systemIOFileTypeSymbol = WellKnownTypes.SystemIOFile(compilation);
+
+                    if (systemIOFileTypeSymbol != null)
+                    {
+                        foreach (var targetMethodName in SystemIOFileMethodMetadataNames)
+                        {
+                            builder.AddRange(systemIOFileTypeSymbol.GetMembers(targetMethodName).Cast<IMethodSymbol>());
+                        }
+                    }
+
+                    var systemReflectionAssemblyTypeSymbol = WellKnownTypes.SystemReflectionAssembly(compilation);
+
+                    if (systemReflectionAssemblyTypeSymbol != null)
+                    {
+                        foreach (var targetMethodName in SystemReflectionAssemblyMethodMetadataNames)
+                        {
+                            builder.AddRange(systemReflectionAssemblyTypeSymbol.GetMembers(targetMethodName).Cast<IMethodSymbol>());
+                        }
+                    }
+
+                    var dangerousMethodSymbols = builder.ToImmutable();
+
+                    if (dangerousMethodSymbols.Length == 0)
+                    {
+                        return;
+                    }
+
+                    compilationStartAnalysisContext.RegisterSymbolAction(
+                        (SymbolAnalysisContext symbolAnalysisContext) =>
+                        {
+                            var classSymbol = symbolAnalysisContext.Symbol as INamedTypeSymbol;
+
+                            if (!classSymbol.GetAttributes().Any(s => s.AttributeClass.Equals(serializableAttributeTypeSymbol)))
+                            {
+                                return;
+                            }
+                            
+                            foreach (var member in classSymbol.GetMembers())
+                            {
+                                if (member.Kind == SymbolKind.Method)
+                                {
+                                    var methodSymbol = member as IMethodSymbol;
+
+                                    if (methodSymbol.GetAttributes().Any(s => attributeTypeSymbols.Contains(s.AttributeClass))
+                                        || methodSymbol.MetadataName == "OnDeserialization")
+                                    {
+                                        var parameters = methodSymbol.GetParameters();
+
+                                        if (parameters == null
```

Comment:
It looks like `IDeserializationCallback.OnDeserialization(object)` takes an object as its parameter.  Is that what the `methodSymbol.MetadataName == "OnDeserialization"` is checking above?

Also, if so, may want to see if the OnDeserialization is implementing that interface's method.